File: Fonction/Accepte-Cylindre.py
import math

# Définition de pi
pi = math.pi

# Constantes
volume_minimum = 450
rayon_minimum = 7

# Volume d’un cylindre
def calcul_volume_cylindre(rayon, hauteur):
    return pi * rayon * rayon * hauteur

# Vérifie si accepté
def accepte_cylindre(rayon, hauteur):
    return (calcul_volume_cylindre(rayon, hauteur) > volume_minimum
            and rayon > rayon_minimum)

# Message utilisateur
def informe(accepte):
    return "Cylindre adapté pour l’expérience" if accepte else "Cylindre rejeté"

# Fonction globale
def filtre_cylindre(rayon, hauteur):
    return informe(accepte_cylindre(rayon, hauteur))

# Tests
print(informe(accepte_cylindre(7, 3)))
print(filtre_cylindre(7, 3))

print(informe(accepte_cylindre(2, 3)))
print(filtre_cylindre(2, 3))

File: Fonction/Accepte-Cylindre.php
<?php
// Définition de pi
$pi = pi();

// Constantes
$volume_minimum = 450;
$rayon_minimum = 7;

// Volume d’un cylindre
function calculVolumeCylindre($rayon, $hauteur) {
    global $pi;
    return $pi * $rayon * $rayon * $hauteur;
}

// Vérifie si accepté
function accepteCylindre($rayon, $hauteur) {
    global $volume_minimum, $rayon_minimum;

    return (calculVolumeCylindre($rayon, $hauteur) > $volume_minimum && $rayon > $rayon_minimum);
}

// Message utilisateur
function informe($accepte) {
    return $accepte
        ? "Cylindre adapté pour l’expérience"
        : "Cylindre rejeté";
}

// Fonction globale
function filtre_cylindre($rayon, $hauteur) {
    return informe(accepteCylindre($rayon, $hauteur));
}
// Tests
echo informe(accepteCylindre(7, 3)) . PHP_EOL;
echo filtre_cylindre(7, 3) . PHP_EOL;

echo informe(accepteCylindre(2, 3)) . PHP_EOL;
echo filtre_cylindre(2, 3) . PHP_EOL;

echo informe(accepteCylindre(7, 10)) . PHP_EOL;
echo filtre_cylindre(7, 10) . PHP_EOL;
?>